src/viewport.cpp

```diff
@@ -1404,11 +1405,23 @@ static bool ViewportSortParentSpritesChecker()
 	return true;
 }
 
+static int CDECL CompareParentSprites(ParentSpriteToDraw * const *psd, ParentSpriteToDraw * const *psd2)
+{
+	const ParentSpriteToDraw *ps = *psd;
+	const ParentSpriteToDraw *ps2 = *psd2;
+	return ps->xmin - ps2->xmin;
+}
+
 /** Sort parent sprites pointer array */
 static void ViewportSortParentSprites(ParentSpriteToSortVector *psdv)
 {
 	auto psdvend = psdv->end();
 	auto psd = psdv->begin();
+	auto psdMax = psd;
+
+	/* pre-sort by xmin in ascending order */
+	QSortT(psdv->data(), psdv->size(), CompareParentSprites);
```

Comment:
We are trying to move to something closer to modern C++, thus something with `std::sort` would be preferable.